packages/core/src/analytics/braintree-connect-tracker/braintree-connect-tracker.ts

```diff
@@ -0,0 +1,187 @@
+import {
+    BraintreeConnect,
+    BraintreeConnectApmSelectedEventOptions,
+    BraintreeConnectEmailEnteredEventOptions,
+    BraintreeConnectEventCommonOptions,
+    BraintreeConnectOrderPlacedEventOptions,
+    isBraintreeConnectWindow,
+} from '@bigcommerce/checkout-sdk/braintree-utils';
+import { PaymentMethodClientUnavailableError } from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { CheckoutService } from '../../checkout';
+
+import BraintreeConnectTrackerService from './braintree-connect-tracker-service';
+
+export default class BraintreeConnectTracker implements BraintreeConnectTrackerService {
+    private shouldTriggerEventSubmittedEvent = true;
+    private selectedPaymentMethodId = 'braintreeacceleratedcheckout';
+
+    constructor(private checkoutService: CheckoutService) {}
+
+    customerPaymentMethodExecuted() {
+        if (this.shouldTrackEvent()) {
+            this.shouldTriggerEventSubmittedEvent = false;
+
+            this.trackEmailSubmitted(false);
+        }
+    }
+
+    trackStepViewed(step: string) {
+        if (
+            this.shouldTrackEvent() &&
+            this.shouldTriggerEventSubmittedEvent &&
+            step !== 'customer'
+        ) {
+            this.shouldTriggerEventSubmittedEvent = false;
+
+            this.trackEmailSubmitted(true);
+        }
+    }
+
+    paymentComplete() {
+        if (this.shouldTrackEvent()) {
+            if (this.shouldTriggerEventSubmittedEvent) {
+                this.trackEmailSubmitted(true);
+            }
+
+            this.trackOrderPlaced(this.selectedPaymentMethodId);
+        }
+    }
+
+    selectedPaymentMethod(methodId: string): void {
+        if (this.shouldTrackEvent() && methodId) {
+            this.selectedPaymentMethodId = methodId;
+
+            this.trackApmSelected(methodId, false);
+        }
+    }
+
+    walletButtonClick(methodId: string) {
+        if (this.shouldTrackEvent() && methodId) {
+            this.selectedPaymentMethodId = methodId;
+
+            this.trackApmSelected(methodId, true);
+        }
+    }
+
+    /**
+     *
+     * Common
+     *
+     */
+    private shouldTrackEvent() {
+        return isBraintreeConnectWindow(window) && window.braintreeConnect.events;
+    }
+
+    private getBraintreeConnectEventsOrThrow(): BraintreeConnect['events'] {
```

Comment:
```suggestion
    private _getBraintreeConnectEventsOrThrow(): BraintreeConnect['events'] {
```

Reply:
And similar for other private methods please.

Reply:
Is it possible to add eslint rule to avoid such "mistakes"?